library/psa_crypto_its.h

```diff
@@ -142,4 +142,8 @@ psa_status_t psa_its_get_info(psa_storage_uid_t uid,
  */
 psa_status_t psa_its_remove(psa_storage_uid_t uid);
 
+#ifdef __cplusplus
+}
```

Comment:
Thanks! We have a test for this (`programs/test/cpp_dummy_build.cpp`) but it only covers `include/mbedtls/*.h`, not `include/psa/*.h` or `library/*.h`. Arguably not all the `library/*.h` headers are intended to be used in C++ (most are only for building the library itself, `psa_crypto_its.h` is a bit of an exception), so I we might not want to include all of them in `cpp_dummy_build.cpp`. But I think it would make sense to add
```
#include "../library/psa_crypto_its.h"
```
(not very clean, but easier that messing around with the include path in both make and cmake). And also `psa/crypto.h` and `psa/crypto_se_driver.h` since they're missing.

Reply:
If I understand the setup correctly, `psa_crypto_its.h` is still only used internally. External implementations of the ITS interface are expected to self-supply a `psa/internal_trusted_storage.h` file (ref. psa_crypto_storage.c).

I just wanted to fix an inconsistency here. Either it's this fix, or remove the CPP guard in this file altogether if it's never meant to be compiled with cpp.

Reply:
Ah, indeed, we require an external implementation to provide its own header.

The commit that created the file (601bd53b80fdd86e42345fdc1394850d2e64540e) states

> Drop __cplusplus support which we don't need.

but I forgot to remove one occurrence. So we intended not to have a guard. But it doesn't really matter as long as it's consistent.